tfexec/terraform_workspace_show.go

```diff
@@ -0,0 +1,25 @@
+package tfexec
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+)
+
+// tfWorkspaceRe is a pattern to parse outputs from terraform workspace show.
+var tfWorkspaceRe = regexp.MustCompile(`^[a-zA-Z\d_-]{1,90}`)
```

Comment:
I couldn't find any docs for a valid workspace name.
Could you tell me where this regular expression came from?

Reply:
I found it here : https://www.terraform.io/docs/cloud/workspaces/naming.html

> Workspace names need to be 90 characters or less and can only include letters, numbers, -, and _.

Looking online that doesn't actually seem to be correct and the actual expectation is that the workspace name should be url-safe. (see: https://github.com/hashicorp/terraform/issues/24564).

All things considered I don't think it is useful to implement this check as it is already handled by terraform so I have simply removed it.

Reply:
It's a common pitfall for workspace. The word `workspace` has two meanings in Terraform. That is, one is a workspace for Terraform Cloud and the other is a workspace for Terraform CLI. The doc you liked is for Terraform Cloud. It's actually not related to `terraform workspace command`.

From the error message, I found the validation for `terraform workspace new`. It only requires a URL safe string.
https://github.com/hashicorp/terraform/blob/v1.0.3/internal/command/workspace_command.go#L42-L47

However, I agree that we don't need duplicated validation here by ourselves.